### ISSUE TITLE
Ability to set pod security for acorn-system

### ISSUE THREAD

**Issue by nilathedragon** [acorn-io/acorn#912]
In a cluster where the default pod security is set to restricted, acorn cannot install successfully. This can be temporarily solved by setting the `pod-security.kubernetes.io/enforce: baseline` label on the namespace `acorn-system`. However, this label is quickly overwritten by some acorn component causing any further deployments in that namespace to fail again. 

It would be great if the label could be configured either by default or set similarly as the pod security for app deployments.

**_Sidenote_**
I assume it's acorn overwriting it, since it's the only namespace that behaves like this. Probably here? (no idea really) https://github.com/acorn-io/acorn/blob/main/pkg/controller/data.go#L13

**Comment by cjellick** [comment]
Do the existing install flags around this not do the trick?
```
$ acorn install --help
Install and configure acorn in the cluster

Usage:
  acorn install [flags]


acorn install

Flags:
...
      --pod-security-enforce-profile string   The name of the PodSecurity profile to set (default baseline)
      --set-pod-security-enforce-profile      Set the PodSecurity profile on created namespaces (default true)
```

**Comment by nilathedragon** [comment]
I think those arguments only apply to the deployed app namespaces and not the `acorn-system` namespace.
I tested again with both arguments set to be sure, and could not see a change in `acorn-system`.

This is the configuration in the config map I used to test (redacted domains):
`{"ingressClassName":null,"clusterDomains":["<my domain>"],"letsEncrypt":"disabled","letsEncryptEmail":"","letsEncryptTOSAgree":null,"setPodSecurityEnforceProfile":true,"podSecurityEnforceProfile":"baseline","defaultPublishMode":"defined","internalClusterDomain":"","acornDNS":"disabled","acornDNSEndpoint":null,"autoUpgradeInterval":null}`

I also restarted the controller to make sure it was applied.

**Comment by nilathedragon** [comment]
I've gotten curious and tested a bit more:

I built a modified acorn image and making the following change. I can now see the controller setting the label correctly every time it starts. So I don't think there is currently a way to set this label properly without it being overwritten on startup.

```go
func (c *Controller) initData(ctx context.Context) error {
	err := c.apply.Ensure(ctx, &corev1.Namespace{
		ObjectMeta: metav1.ObjectMeta{
			Name: system.Namespace,
			Labels: map[string]string{
				"pod-security.kubernetes.io/enforce": "privileged",
			},
		},
	})
	if err != nil {
		return err
	}
	if err := config.Init(ctx, c.client); err != nil {
		return err
	}
	return buildkit.SyncBuildkitPod(ctx, c.client)
}
```

**Comment by cjellick** [comment]
ok, interesting. we'll get this fixed.

**Comment by ibuildthecloud** [comment]
We can't apply `pod-security.kubernetes.io/enforce: baseline` to `acorn-system` because we run pods there that need NET_ADMIN capability and also the buildkitd pod runs as privileged.

**Comment by nilathedragon** [comment]
> We can't apply `pod-security.kubernetes.io/enforce: baseline` to `acorn-system` because we run pods there that need NET_ADMIN capability and also the buildkitd pod runs as privileged.

You are correct, sorry for that initial mistake! I noticed that after reporting. But even then this issue would still apply since the namespace has no label by default, which won't work in a restricted-by-default environment

**Comment by cjellick** [comment]
I understood this to mean we'd want to support, like
```
acorn install --system-pod-security-enforce-profile privileged
```
eh?
